cpp/src/arrow/table.h

```diff
@@ -63,6 +63,17 @@ class ARROW_EXPORT Table {
                                      const std::vector<std::shared_ptr<Array>>& arrays,
                                      int64_t num_rows = -1);
 
+  /// \brief Create an empty Table of a given schema
+  ///
+  /// The output Table will be created from a ChunkedArrayVector
+  /// with DataTypes from the schema.
```

Comment:
Hmm, the fact that a ChunkedArrayVector is used internally is irrelevant. Perhaps shorten this to:
"""The Table will be created with a single empty chunk per column""".